src/main/grammars/RustParser.bnf

```diff
@@ -1001,8 +1001,17 @@ MacroMatchingTree ::=  '(' MacroMatchingTree* ')'
                      | (MacroPatternSimpleMatching | MacroPatternComplexMatching)+
                      | << placeholderMacroToken >>+
 
-MacroPatternComplexMatching ::= '$' '(' MacroMatchingTree* ')' <<tokenAfterComplexPattern>>* ('*' | '+')
-MacroPatternSimpleMatching ::= '$' MacroBinding ':' identifier
+MacroPatternComplexMatchingBody ::= '(' MacroMatchingTree* ')' {
+  pin = 1
+}
+MacroPatternComplexMatching ::= '$' MacroPatternComplexMatchingBody <<tokenAfterComplexPattern>>* ('*' | '+')  {
+  implements = [ "org.rust.lang.core.psi.ext.RsNamedElement" ]
```

Comment:
Only simple atomic bindings like `$foo:ty` should be `RsNamedElements` and `RsReferences`, you don't need to mach the whole complex matching structure during resolve.